ethers-core/src/types/block.rs

```diff
@@ -89,6 +89,49 @@ pub struct Block<TX> {
     pub epoch_snark_data: Option<EpochSnarkData>,
 }
 
+// ref https://eips.ethereum.org/EIPS/eip-1559
+pub const ELASTICITY_MULTIPLIER: U256 = U256([2u64, 0, 0, 0]);
+// max base fee delta is 12.5%
+pub const BASE_FEE_MAX_CHANGE_DENOMINATOR: U256 = U256([8u64, 0, 0, 0]);
+
+impl<TX> Block<TX> {
+    /// The target gas usage as per EIP-1559
+    pub fn gas_target(&self) -> U256 {
+        self.gas_limit / ELASTICITY_MULTIPLIER
+    }
+
+    /// The next block's base fee, it is a function of parent block's base fee and gas usage.
+    /// Reference: https://eips.ethereum.org/EIPS/eip-1559
+    pub fn next_block_base_fee(&self) -> Option<U256> {
+        let target_usage = self.gas_target();
+        if let None = self.base_fee_per_gas {
+            return None
+        }
+
+        let base_fee_per_gas = self.base_fee_per_gas.unwrap();
+
+        if self.gas_used > target_usage {
```

Comment:
this if elsif else can optionally be rewritten with a `match self.gas_used.cmp(&target_usage)`

Reply:
Is it supposed to be like:
```rs
match self.gas_used.cmp(&target_usage) {
    Ordering::Less => {...},
    Ordering::Greater => {...},
    Ordering::Equal => {...},
}
```

Reply:
yes!
usually, clippy complains if that's an if elseif else